src/Microsoft.ML.Data/Commands/ScoreCommand.cs

```diff
@@ -225,13 +226,20 @@ public static class ScoreUtils
     {
         public static IDataScorerTransform GetScorer(IPredictor predictor, RoleMappedData data, IHostEnvironment env, RoleMappedSchema trainSchema)
         {
-            var sc = GetScorerComponentAndMapper(predictor, null, data.Schema, env, out var mapper);
-            return sc.CreateInstance(env, data.Data, mapper, trainSchema);
+            var sc = GetScorerComponentAndMapper(predictor, null, data.Schema, env, null, out var mapper);
+            return sc.CreateComponent(env, data.Data, mapper, trainSchema);
         }
 
-        public static IDataScorerTransform GetScorer(SubComponent<IDataScorerTransform, SignatureDataScorer> scorer,
-            IPredictor predictor, IDataView input, string featureColName, string groupColName,
-            IEnumerable<KeyValuePair<RoleMappedSchema.ColumnRole, string>> customColumns, IHostEnvironment env, RoleMappedSchema trainSchema)
+        public static IDataScorerTransform GetScorer(
+            IComponentFactory<IDataView, ISchemaBoundMapper, RoleMappedSchema, IDataScorerTransform> scorer,
```

Comment:
>IComponentFactory<IDataView, ISchemaBoundMapper, RoleMappedSchema, IDataScorerTransform> [](start = 12, length = 88)

can you define something like using TScorerFactory = IComponentFactory<IDataView, ISchemaBoundMapper, RoleMappedSchema, IDataScorerTransform> and use it across this file?
see  line 17 in MetaMulticlassTrainer.

Reply:
Done.